tests/tests/ecaddg2_test.rs

```diff
@@ -0,0 +1,427 @@
+mod test_utils;
+use test_utils::{era_call, parse_call_result, write_ecaddg2_gas_result};
+use zksync_web3_rs::types::{Address, Bytes, H160};
+
+pub const ECADD_G2_PRECOMPILE_ADDRESS: Address = H160([
+    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+    0x00, 0x00, 0x00, 0x0A,
+]);
+
+const EXECUTION_REVERTED: &str =
+    "(code: 3, message: execution reverted, data: Some(String(\"0x\")))";
+
+#[tokio::test]
+async fn ecadd_g2_valid_1() {
```

Comment:
Is there a reference from where these numbers and tests come from?
I can't really understand what's happening here.